Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/core/repository/FamilyTestRepository.kt

```diff
@@ -1,76 +1,125 @@
 package de.rki.coronawarnapp.familytest.core.repository
 
-import de.rki.coronawarnapp.coronatest.TestRegistrationRequest
+import de.rki.coronawarnapp.coronatest.qrcode.CoronaTestQRCode
 import de.rki.coronawarnapp.coronatest.type.TestIdentifier
 import de.rki.coronawarnapp.familytest.core.model.FamilyCoronaTest
-import de.rki.coronawarnapp.util.coroutine.AppScope
-import kotlinx.coroutines.CoroutineScope
+import de.rki.coronawarnapp.familytest.core.model.markBadgeAsViewed
+import de.rki.coronawarnapp.familytest.core.model.markDccCreated
+import de.rki.coronawarnapp.familytest.core.model.markViewed
+import de.rki.coronawarnapp.familytest.core.model.recycle
+import de.rki.coronawarnapp.familytest.core.model.restore
+import de.rki.coronawarnapp.familytest.core.model.updateResultNotification
+import de.rki.coronawarnapp.familytest.core.storage.FamilyTestStorage
+import de.rki.coronawarnapp.util.TimeStamper
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
 import javax.inject.Inject
+import javax.inject.Singleton
 
+@Singleton
 class FamilyTestRepository @Inject constructor(
```

Comment:
FamilyTestRepository is NOT thread-safe!

Reply:
Please elaborate more? you mean because the flow is not shared ?